spec/models/product_user_creator_spec.rb

```diff
@@ -0,0 +1,43 @@
+require "spec_helper"
+
+describe ProductUserCreator do
+  subject do
+    -> { ProductUserCreator.create(user: user, product: product, approver: approver) }
```

Comment:
This is an interesting idiom. Where'd you get the idea?

I feel like it might be easier to understand to actually define a method here called `create_product_user`, that you can use when you're testing the side effects, and you can keep `let(:product_user) { create_product_user }` as an alias when you're looking for actual references to the object.